### ISSUE TITLE
[Suggestion] Change of the default amount of RAM to allocate

### ISSUE THREAD

**Issue by La1eon** [Kron4ek/minecraft-vortex-launcher#14]
To gather more information about the topic I'm gonna write about here, I did several simple tests. If you are interested about, you can see the results at the very bottom.

After seeing the results of these tests it seems to me that the current default amount of RAM to allocate, which is 700 MB, is a bit too little for Minecraft versions 1.13 and above. What do you think about increasing it to 1024 MB? Or maybe even to 2048 MB (but it depends on your preferences)?

For comparison the official Minecraft launcher was proposing by default 1 GB of RAM for years. Some time ago it changed the amount to 2 GB, probably due to the fact that Minecraft versions 1.13 and above need noticeably more RAM than version 1.12.2. Furthermore, most of "normal" computer users have 4 or 8 GB of RAM (data from [here](https://data.firefox.com/dashboard/hardware), I mostly take such data from there), so 1 or 2 GB for Minecraft doesn't seem to be too much (at least for a desktop application that utilizes JVM :p).

---

To find out how much of RAM individual versions take, I tested it on my PC with Windows 8.1 64-bit, 8 GB RAM, JRE 8u221 (64-bit), Vortex Launcher 1.1.9, and "-Xmx1024M" argument, among the 2 different default sets of JVM arguments being passed by this launcher. Every time I was downloading Minecraft files to an empty directory, after that I was launching the game, creating a new classic-type superflat world, changing the game difficulty to "peaceful", looking around to let all chunks load, opening F3 overlay and writing down min and max RAM usage, closing the game, deleting all files (except the launcher .exe file) from the directory, in which the launcher was located, and repeating this schema a dozen times (once for each version, except 1.6.4, 1.13.2, and 1.14.4).

Here are the results:
* 1.6.4 – 220–320 MB (of RAM usage)
* 1.7.10 – 90–190 MB
* 1.8.9 – 130–230 MB
* 1.9.4 – 100–200 MB
* 1.10.2 – 100–200 MB
* 1.11.2 – 100–200 MB
* 1.12.2 – 110–210 MB
* 1.13.2 – I tested it multiple times, and the largest range I got was 270–720 MB (with the new, default set of JVM arguments for versions 1.13 and above); I also noted 250–350 MB of RAM usage with the legacy, default set of JVM arguments for versions below 1.13
* 1.14.4 – also in this case I tested it multiple times, and the largest range I got was 300–730 MB (with the new, default set of JVM arguments for versions 1.13 and above); I also noted 240–340 MB of RAM usage with the legacy, default set of JVM arguments for versions below 1.13

**Comment by Kron4ek** [comment]
I agree that 1 GB is more sane for recent game versions (1.13+) and 2 GB is even better, but there is a reason why i decided to reduce default value to 700 MB. The reason is that some people use 32-bit Java on 64-bit OS (by mistake or ignorance), and, due to limitations of 32-bit processes, allocation of 1 GB memory on 32-bit Java leads to frequent crashes on recent game versions (at least that happens on Windows).

Reducing default memory value is rough yet simple solution, another possible solution is to check what Java architecture is used and offer another default value according to that. If i find a reliable way to detect Java architecture maybe i will change default value.

**Comment by La1eon** [comment]
Oh, OK. Thanks for the explanation. I have one additional question due to my curiosity: doesn't the Java 8 online installer, which is the default one, automatically choose the right version (64-bit for 64-bit OS, and so on)? I know that once I accidentally installed 32-bit JRE 8 on a 64-bit Windows, but I don't remember how it happened.

**Comment by La1eon** [comment]
Thanks for the explanation. I have one additional question due to my curiosity: doesn't the Java 8 online installer, which is the default one, automatically choose the right version (64-bit for 64-bit OS, and so on)? I know that once I accidentally installed 32-bit JRE 8 on a 64-bit Windows, but I don't remember how it happened.

**Comment by Kron4ek** [comment]
Yes, as far as i know the online installer automatically choses architecture according to the OS arch. But i often see people that have wrong Java architecture installed, perhaps they manually download 32-bit offline installer (maybe even not from the official site).
